UI Fixes #79, #81: Back button, Small screen layouts

nostrbuddha opened the pull request:
This PR has fixes for https://github.com/bisq-network/bisq-mobile/issues/81 (temporary patch) and https://github.com/bisq-network/bisq-mobile/issues/79

Summarizing here. Added more in those specific issues:
 * Back button logic generalized to TopBar.
 * Back swipe gesture implemented for iOS. But it's not as good as back gesture in SwiftUI. Details in the issue page.
 * Tested in Pixel 6a and Pixel 2. Made changes to Scaffold, Layout components so now they all look good. (@rodvar Please check and confirm me from your side)

rodvar commented:
rebased branch against main

rodvar commented on shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/SwipeBackIOSNavigationHandler.kt:
```diff
@@ -0,0 +1,66 @@
+// - https://github.com/adrielcafe/voyager/issues/144
+// - https://trycatchdebug.net/news/1426361/native-ios-swipe-back-gesture-in-compose
+@Composable
+fun SwipeBackIOSNavigationHandler(
```
Nice hack Buddha

glad to see the basic kmp Platform example got its real use finally :D 

ideally the `isIos` question should be passed to a presenter instead of the view finding out by calling a global model related getPlatform itself.

rodvar commented on shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/molecules/TopBar.kt:
```diff
@@ -1,36 +1,60 @@
     title: String = "",
     isHome: Boolean = false,
-    navigationIcon: @Composable () -> Unit = {}
+    customBackButton: @Composable (() -> Unit)? = null
```
excellent

rodvar reviewed (changes requested):
UI wise looks great - functionality wise needs more work:

 - bug: (100% reproducible) after the back arrow has been rendered at least once, clicking on the title will blank all the UI (see screenshot)
![Screenshot from 2024-11-30 10-39-49](https://github.com/user-attachments/assets/acb11f74-2e93-404d-bafe-0d096712cc56)
  - bug: sometimes the back arrow won't show up (I could only repro this once, could't figure out the steps yet)

I've also put some comments on the code.
iPhone swipe works!

nostrbuddha commented:
Missed that back button white screen issues!
Fixed it now and also moved isIOS() to App/MainPresenter.

Tried replicating back button not appearing. Couldn't make it happen! Can you please try this again and give me any hints?

rodvar reviewed (approved):
great job buddha
